components/raftstore/src/store/peer_storage.rs

```diff
@@ -767,12 +799,191 @@ where
         Ok(())
     }
 
+    pub fn set_async_fetch_res(&mut self, res: Option<RaftlogFetchResult>) {
+        let in_fetching = match self.async_fetch_res.borrow().as_ref() {
+            r @ Some(RaftlogFetchResult::Fetched { .. }) => {
+                warn!(
```

Comment:
I think it's a normal behavior. For example, it fallbacks to catch up logs by snapshot.